src/System.Text.RegularExpressions/src/System/Text/RegularExpressions/RegexCompiler.cs

```diff
@@ -113,12 +113,12 @@ internal abstract class RegexCompiler
         private const int Uniquecount = 10;
         private const int LoopTimeoutCheckCount = 2048; // A conservative value to guarantee the correct timeout handling.
 
-        private static FieldInfo RegexRunnerField(string fieldname)
+        private static FieldInfo? RegexRunnerField(string fieldname)
```

Comment:
Does this actually return null, assuming the specified field exists?  If we expect it to always exist, seems better to change this to instead return `FieldInfo` and just Debug.Assert that GetField doesn't return null.

Reply:
Good point. We can assume existence is guaranteed by our test coverage, right?